src/functions-and-arrays.js

```diff
@@ -1,73 +1,175 @@
 // Iteration #1: Find the maximum
+function maxOfTwoNumbers(number1, number2) {
+    if (number1 > number2) {
+        return number1;
+    } else {
+        return number2;
+    }
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(words) {
+    if (words.length > 0) {
+        return words.sort(function(a, b) { return b.length - a.length })[0];
+    } else {
+        return null
+    }
+}
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(numbers) {
+    if (numbers.length === 0) {
+        return 0
+    } else {
+        let total = 0;
+        for (i = 0; i < numbers.length; ++i) {
+            total += numbers[i];
+        }
+        return total;
+    }
+}
+
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(numbersAvg) {
+    if (numbersAvg.length === 0) {
+        return null
+    } else {
+        let total = 0;
+        for (let i = 0; i < numbersAvg.length; i++) {
+            total += numbersAvg[i];
+        }
+        let avg = total / numbersAvg.length;
+        return avg
+    }
+}
+
```

Comment:
You can replace line 48-53 with `return sumNumbers(numbersAvg)/numbersAvg.length`